src/Shared/PackageManagers/PackageManagerFactory.cs

```diff
@@ -76,6 +77,57 @@ public static BaseProjectManager CreateBaseProjectManager(string destinationDire
 
             return null;
         }
+        
+        /// <summary>
+        /// Get an appropriate project manager for package given its PackageURL, with a provider.
+        /// </summary>
+        /// <param name="purl">The <see cref="PackageURL"/> for the package to create the project manager for.</param>
+        /// <param name="nuGetProvider">The <see cref="INuGetProvider"/> for to make calls with.</param>
+        /// <param name="httpClientFactory"> The <see cref="IHttpClientFactory"/> for the project manager to use for making Http Clients to make web requests.</param>
+        /// <param name="destinationDirectory">The directory to use to store any downloaded packages.</param>
+        /// <returns> BaseProjectManager object </returns>
+        public static BaseProjectManager? CreateProjectManagerWithProvider(PackageURL purl, INuGetProvider nuGetProvider, IHttpClientFactory? httpClientFactory = null, string? destinationDirectory = null)
+        {
+            if (projectManagers.Count == 0)
+            {
+                projectManagers.AddRange(typeof(BaseProjectManager).Assembly.GetTypes()
+               .Where(type => type.IsSubclassOf(typeof(BaseProjectManager))));
+            }
+            // Use reflection to find the correct package management class
+            Type? downloaderClass = projectManagers
+               .Where(type => type.Name.Equals($"{purl.Type}ProjectManager",
+                                               StringComparison.InvariantCultureIgnoreCase))
+               .FirstOrDefault();
+
+            if (downloaderClass != null)
+            {
+                if (downloaderClass != typeof(NuGetProjectManager))
```

Comment:
If its not nuget we just shortcut this method and ignore the provider? This doesn't seem correct to me.

Reply:
I see why you are doing this now based on the way you are calling it but I don't think this is the correct way.

Reply:
Also should be resolved in the new PR. I no longer do any checks for if it is a NuGetProjectManager or not.